libnetwork/drivers/bridge/port_mapping_linux.go

```diff
@@ -869,6 +876,59 @@ func setPerPortForwarding(b portBinding, ipv iptables.IPVersion, bridgeName stri
 	return nil
 }
 
+// filterPortByInputIface adds a couple of iptables rules to accept packets
+// destined to a mapped port with a dest addr that matches the interface
+// they're received on. If that doesn't match, the second rule drop them
+// unceremoniously.
+//
+// These rules will block rogue hosts that try to access a mapped port while
+// they aren't part of the L2 segment where the mapped port is exposed.
+// For instance, if HostIP == 127.0.0.1, no ingress should come from anything
+// but lo.
+func (n *bridgeNetwork) filterPortByInputIface(b portBinding, enable bool) error {
+	hostIP := b.childHostIP
+	if b.HostPort == 0 {
+		// Direct routing mode is used, we can't filter based on the input iface.
+		return nil
+	}
+
+	// DOCKER_DISABLE_INPUT_IFACE_FILTERING can be used as an escape hatch if
+	// this filtering doesn't work out well for some users.
+	if v, _ := strconv.ParseBool(os.Getenv("DOCKER_DISABLE_INPUT_IFACE_FILTERING")); v {
+		log.G(context.TODO()).Info("DOCKER_DISABLE_INPUT_IFACE_FILTERING is set, skipping input iface filtering.")
```

Comment:
Thanks! I'd even be ok with `Warn` for this if we consider it "bad", but Info works for me.

Reply:
That's now a `Warn`.